### DISCUSSION TITLE
Window-scoped binds and no-op dispatcher

### DISCUSSION THREAD

**Discussion by keliss** [hyprwm/Hyprland#7562] [General]
Hello, here's my use case: I have a laptop where the `PgUp` and `PgDn` buttons are too close to the arrow keys, so I end up misclicking them often. I wanted to disable them in all apps except where they are actually useful (like browser), but there's neither a way to specify a window rule to do that, nor a no-op dispatcher to use in binds if I'm trying to work around this.
I ended up writing a script that listens on the socket and issues `hyprctl keyword` every time an active window changes to bind/unbind necessary keys, where the `exec` dispatcher serves as a no-op, which is not ideal - a bit messy.
So the idea is to allow binds in window rules (or at least expose callbacks like `onActive` to simplify scripting), and add a new no-op dispatcher. WDYT?

**Comment by sungyoonc** [comment]
For your initial use case, you could try [rvaiya/keyd](https://github.com/rvaiya/keyd) with its [application specific remapping](https://github.com/rvaiya/keyd#application-specific-remapping-experimental)

**Reply by keliss** [reply]
Thank you, will give it a try. But I'd rather not install a whole bunch of things that handle key bindings, I'm afraid it'll quickly become unmanageable. Hyprland on its own should be enough for that given it already takes that role upon itself.
